exporter/exporterhelper/internal/queuebatch/config.go

```diff
@@ -46,6 +47,16 @@ type Config struct {
 
 	// BatchConfig it configures how the requests are consumed from the queue and batch together during consumption.
 	Batch configoptional.Optional[BatchConfig] `mapstructure:"batch"`
+
+	// MetadataKeys is a list of client.Metadata keys that will be used to partition
+	// the data into batches. If this setting is empty, a single batcher instance
+	// will be used. When this setting is not empty, one batcher will be used per
+	// distinct combination of values for the listed metadata keys.
+	//
+	// Empty value and unset metadata are treated as distinct cases.
+	//
+	// Entries are case-insensitive. Duplicated entries will trigger a validation error.
+	MetadataKeys []string `mapstructure:"metadata_keys"`
```

Comment:
In your implementation, this field is ignored when using WithQueue. That seems very surprising, and I think undesirable. Can we make it so WithQueue respects it?

Reply:
> I kept the otlphttp exporter as [xexporterhelper.WithQueueBatch(oCfg.QueueConfig, qbs))](https://github.com/open-telemetry/opentelemetry-collector/pull/14139/files#diff-41aa0d4e53d41b553d1e0a4c279130b16585a0ef48c7f934237e6a0081529274R129)

Why?